File: Client_trunk2/Assets/Scripts/Spells/Effects/EffectLighting.cs
using UnityEngine;
using System.Collections;
namespace SPELL
{
	/// <summary>
	/// 播放特效声音的技能效果
	/// </summary>
	public class EffectLighting : SpellEffect
	{		
		public string bindingObjectPath = "";                      
		public string effect = "";
		public string sound = "";
		public bool isBind = false;
        public float posOffset = 0;
        public bool setRotation = false; //强制朝向

        public override void Init(DataSection.DataSection dataSection)
		{
			base.Init(dataSection);
			bindingObjectPath = dataSection.readString("param1");
			effect = dataSection.readString("param2");
			sound = dataSection.readString("param3");
			isBind = dataSection.readBool("param4");
            posOffset = dataSection.readFloat("param5");
            setRotation = dataSection.readBool("param6");
        }
		
		/// <summary>
		/// caster do something to dst use by spell
		/// </summary>
		/// <param name="spell">Spell.</param>
		/// <param name="src">spell caster.</param>
		/// <param name="dst">spell receiver.</param>
		public override void Cast(AvatarComponent src, AvatarComponent dst, SpellEx spell, SpellTargetData targetData)
		{
			Transform transf = dst.myTransform;
			if (bindingObjectPath != "")
			{
				transf = transf.FindChild(bindingObjectPath);
				if (!transf)
					transf = dst.myTransform;
			}

            EffectComponent eComponent = null;
            // 播放光效
            if (effect != "")
            {
                if (isBind)
                    eComponent = dst.effectManager.AddEffect(effect, transf);
                else
                {
                    Vector3 pos = transf.position + transf.forward * posOffset;
                    eComponent = dst.effectManager.AddEffect(effect, pos);
                }

                //强制特效朝向    
                if (setRotation)
                {
                    eComponent.gameObject.transform.rotation = dst.myTransform.rotation;
                }
                    
            }

            // 播放声音
            if (sound != "")
                AudioManager.Instance.SoundPlay(sound);

            BulletMotionSkill bs = eComponent.gameObject.GetComponent<BulletMotionSkill>();
            if (bs)
                bs.Init(src);

        }
		
	}
}

File: Server_trunk/res/scripts/cell/Spell/Effects/EffectLighting.py
# -*- coding: utf-8 -*-
#
"""
"""
from KBEDebug import *
from ..Spell import SpellEffect

class EffectLighting(SpellEffect):
	"""
	光效播放
	"""
	def init(self, dataSection):
		"""
		"""
		SpellEffect.init(self, dataSection)
	
	def cast(self,src, dst , spell):
		"""
		src向dst施放一个效果
		@param src: Entity; 施法者
		@param dst: Entity; 受术者
		@param spell: instance of Spell; 表示是由哪个技能触发的效果
		@return: None
		"""
		# 允许没有src
		srcID = -1
		if src is not None:
			srcID = src.id

		dst.allClients.seeSpellEffectFS( self.id, srcID )
